lib/logger/lib/logger/legacy_handler.ex

```diff
@@ -5,6 +5,17 @@ defmodule Logger.LegacyHandler do
 
   @internal_keys [:counter]
 
+  def load_log_level do
+    # TODO: Deprecate if level is warning.
+    # TODO: Consider deprecating logger.level altogether in favor of kernel.logger_level.
+    level = case Application.fetch_env!(:logger, :level) do
+      :warn -> :warning
+      level -> level
+    end
+
+    :ok = :logger.set_primary_config(:level, level)
```

Comment:
As earlier, we need to find a way to handle situation when `kernel.logger_level` and `logger.level` are set and which one should be more important in such situation.

Reply:
Good point. Maybe we should allow logger.level to be unset and then we default to kernel.logger_level. Let's tackle this before merging.

Reply:
Unfortunately only way I see for doing that is to introduce breaking change:

```elixir
case Application.fetch_env(:logger, :level) do
  {:ok, :warn} -> :warning
  {:ok, level} -> level
  :error -> Application.fetch_env!(:kernel, :logger_level)
end
```

But that will set level to `:notice` by default instead of current `:debug`.